packages/transport/src/modules/messages/MessageController.ts

```diff
@@ -287,26 +287,45 @@ export class MessageController extends TransportController {
     @log()
     public async sendMessage(parameters: ISendMessageParameters): Promise<Message> {
         const parsedParams = SendMessageParameters.from(parameters);
+
         if (!parsedParams.attachments) parsedParams.attachments = [];
 
         const secret = await CoreCrypto.generateSecretKey();
         const serializedSecret = secret.serialize(false);
         const addressArray: ICoreAddress[] = [];
         const envelopeRecipients: MessageEnvelopeRecipient[] = [];
+        const deletedPeers: string[] = [];
+        const peersWithMissingOrInactiveRelationship: string[] = [];
         for (const recipient of parsedParams.recipients) {
-            const relationship = await this.relationships.getActiveRelationshipToIdentity(recipient);
-            if (!relationship) {
-                throw TransportCoreErrors.messages.missingOrInactiveRelationship(recipient.toString());
+            const relationship = await this.relationships.getRelationshipToIdentity(recipient);
+            if (
+                relationship &&
+                (relationship.status === RelationshipStatus.Terminated || relationship.status === RelationshipStatus.Active) &&
+                relationship.peerDeletionInfo?.deletionStatus !== "Deleted"
+            ) {
+                const cipherForRecipient = await this.secrets.encrypt(relationship.relationshipSecretId, serializedSecret);
+                envelopeRecipients.push(
+                    MessageEnvelopeRecipient.from({
+                        address: recipient,
+                        encryptedKey: cipherForRecipient
+                    })
+                );
+                addressArray.push(recipient);
             }
+            if (!(relationship && (relationship.status === RelationshipStatus.Terminated || relationship.status === RelationshipStatus.Active))) {
+                peersWithMissingOrInactiveRelationship.push(recipient.address);
+            }
+            if (relationship?.peerDeletionInfo?.deletionStatus === "Deleted") {
+                deletedPeers.push(recipient.address);
+            }
```

Comment:
I feel like this section could be simplified, e.g. by restructuring or introducing variables with speaking names for repeated boolean expressions.